zerver/lib/user_groups.py

```diff
@@ -116,7 +116,11 @@ def has_user_group_access(
     if can_edit_all_user_groups:
         return True
 
-    return is_user_in_group(user_group.can_manage_group, user_profile)
+    return user_has_permission_for_group_setting(
+        user_group.can_manage_group,
+        user_profile,
+        NamedUserGroup.GROUP_PERMISSION_SETTINGS["can_manage_group"],
+    )
```

Comment:
This is perfectly readably, but I wonder if this should be calling `user.has_permission()` instead? I'm not sure why we need the two different functions.

Reply:
The `has_permission` function is used for realm settings as of now and then we would either need to pass the group object to that function but only for group settings and not other enum policies or pass `stream` or `group` object and then fetch the setting group from realm, stream or group depending upon the type of setting.

Reply:
So, it just seemed cleaner to have a separate function.